File: backend/hackathon/src/main/java/com/dev_trails/hackathon/DataSeeder.java
package com.dev_trails.hackathon;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
@Component
public class DataSeeder implements CommandLineRunner {
private final PolicyRepository policyRepo;
private final ClaimRepository claimRepo;
private final BillingTransactionRepository billingRepo;
private final RiderRepository riderRepo;
public DataSeeder(PolicyRepository policyRepo, ClaimRepository claimRepo, BillingTransactionRepository billingRepo, RiderRepository riderRepo) {
this.policyRepo = policyRepo;
this.claimRepo = claimRepo;
this.billingRepo = billingRepo;
this.riderRepo = riderRepo;
}
@Override
public void run(String... args) {
if (policyRepo.count() > 0) return;
seedRiders();
seedPolicies();
seedClaims();
seedTransactions();
}
private void seedRiders() {
String[][] data = {
{"Arjun Mehta","+91 98765 43210","Delhi","MZ-DEL-04","Swiggy","28"},
{"Priya Sharma","+91 87654 32109","Mumbai","MZ-MUM-12","Zomato","31"},
{"Rahul Verma","+91 76543 21098","Bangalore","MZ-BLR-07","Swiggy","25"},
{"Sneha Patel","+91 65432 10987","Delhi","MZ-DEL-09","Zepto","29"},
{"Vikram Singh","+91 54321 09876","Hyderabad","MZ-HYD-03","Swiggy","26"},
{"Anita Desai","+91 43210 98765","Chennai","MZ-CHN-05","Zomato","30"},
{"Karan Kapoor","+91 32109 87654","Pune","MZ-PUN-02","Swiggy","27"},
{"Meera Joshi","+91 21098 76543","Delhi","MZ-DEL-04","Dunzo","32"},
{"Deepak Reddy","+91 10987 65432","Hyderabad","MZ-HYD-08","Swiggy","24"},
{"Lakshmi Iyer","+91 09876 54321","Chennai","MZ-CHN-11","Zomato","33"},
};
for (String[] d : data) {
Rider r = new Rider();
r.name = d[0]; r.phone = d[1]; r.city = d[2]; r.zone = d[3]; r.platform = d[4]; r.age = Integer.parseInt(d[5]);
r.walletBalance = 500.0; r.isPolicyActive = true; r.policyTier = "PRO";
riderRepo.save(r);
}
}
private void seedPolicies() {
Object[][] data = {
{"POL-GW-2026-001","Arjun Mehta","MZ-DEL-04 (Connaught Place)","Heat Shield Pro","₹187/day",82,"2026-01-15","+91 98765 43210","arjun.mehta@gmail.com","42 Janpath Road, Connaught Place, New Delhi 110001","1995-08-12","2025-11-20","Gold","NO"},
{"POL-GW-2026-002","Priya Sharma","MZ-MUM-12 (Andheri West)","Rain Guard Plus","₹143/day",71,"2026-02-03","+91 87654 32109","priya.sharma@outlook.com","18/B Versova Link Road, Andheri West, Mumbai 400053","1992-03-25","2025-12-15","Silver","NO"},
{"POL-GW-2026-003","Rahul Verma","MZ-BLR-07 (Koramangala)","Heat Shield Basic","₹129/day",65,"2026-02-20","+91 76543 21098","rahul.verma@yahoo.com","5th Cross, Koramangala 4th Block, Bangalore 560034","1998-11-07","2026-01-10","Bronze","NO"},
{"POL-GW-2026-004","Sneha Patel","MZ-DEL-09 (Karol Bagh)","Heat Shield Pro","₹218/day",91,"2026-01-28","+91 65432 10987","sneha.patel@gmail.com","12 Pusa Road, Karol Bagh, New Delhi 110005","1994-06-18","2025-10-05","Gold","YES"},
{"POL-GW-2026-005","Vikram Singh","MZ-HYD-03 (HITEC City)","Multi-Peril Cover","₹112/day",58,"2026-03-01","+91 54321 09876","vikram.singh@hotmail.com","301 Cyber Towers, HITEC City, Hyderabad 500081","1997-01-30","2026-02-01","Bronze","NO"},
{"POL-GW-2026-006","Anita Desai","MZ-CHN-05 (T. Nagar)","Rain Guard Plus","₹168/day",76,"2026-03-10","+91 43210 98765","anita.desai@gmail.com","78 Usman Road, T. Nagar, Chennai 600017","1993-09-14","2026-01-22","Silver","NO"},
{"POL-GW-2026-007","Karan Kapoor","MZ-PUN-02 (Hinjewadi)","Heat Shield Basic","₹95/day",44,"2026-02-14","+91 32109 87654","karan.kapoor@gmail.com","Blue Ridge Township, Hinjewadi Phase 1, Pune 411057","1996-12-03","2026-01-05","Bronze","NO"},
{"POL-GW-2026-008","Meera Joshi","MZ-DEL-04 (Connaught Place)","Heat Shield Pro","₹201/day",88,"2026-01-05","+91 21098 76543","meera.joshi@outlook.com","15 Barakhamba Road, Connaught Place, New Delhi 110001","1991-04-22","2025-09-18","Platinum","NO"},
{"POL-GW-2026-009","Deepak Reddy","MZ-HYD-08 (Gachibowli)","Multi-Peril Cover","₹125/day",63,"2026-03-18","+91 10987 65432","deepak.reddy@yahoo.com","ISB Road, Gachibowli, Hyderabad 500032","1999-07-09","2026-02-28","Bronze","NO"},
{"POL-GW-2026-010","Lakshmi Iyer","MZ-CHN-11 (Adyar)","Rain Guard Plus","₹139/day",70,"2026-02-28","+91 09876 54321","lakshmi.iyer@gmail.com","22 Gandhi Nagar, Adyar, Chennai 600020","1990-02-16","2026-01-15","Silver","YES"},
};
for (int i = 0; i < data.length; i++) {
Object[] d = data[i];
Policy p = new Policy();
p.policyNumber = (String) d[0]; p.riderId = (long)(i+1); p.riderName = (String) d[1]; p.zone = (String) d[2]; p.plan = (String) d[3];
p.premium = (String) d[4]; p.riskScore = (Integer) d[5]; p.startDate = (String) d[6]; p.phone = (String) d[7]; p.email = (String) d[8];
p.address = (String) d[9]; p.birthdate = (String) d[10]; p.customerSince = (String) d[11]; p.accountTier = (String) d[12];
p.delinquencyStatus = (String) d[13]; p.status = "Active";
policyRepo.save(p);
}
}
private void seedClaims() {
Object[][] data = {
{"CLM-000212","POL-GW-2026-001",1L,"Arjun Mehta","Heat Shield Pro","NO","2026-03-24","Open","HEAT",350.0,"MZ-DEL-04",null},
{"CLM-000198","POL-GW-2026-001",1L,"Arjun Mehta","Heat Shield Pro","NO","2026-03-18","Closed","HEAT",350.0,"MZ-DEL-04","2026-03-18 14:22:16"},
{"CLM-000210","POL-GW-2026-004",4L,"Sneha Patel","Heat Shield Pro","YES","2026-03-24","Open","HEAT",350.0,"MZ-DEL-09",null},
{"CLM-000209","POL-GW-2026-002",2L,"Priya Sharma","Rain Guard Plus","NO","2026-03-24","Open","RAIN",280.0,"MZ-MUM-12",null},
{"CLM-000211","POL-GW-2026-006",6L,"Anita Desai","Rain Guard Plus","NO","2026-03-24","Open","RAIN",280.0,"MZ-CHN-05",null},
{"CLM-000207","POL-GW-2026-010",10L,"Lakshmi Iyer","Rain Guard Plus","NO","2026-03-23","Open","RAIN",280.0,"MZ-CHN-11",null},
{"CLM-000213","POL-GW-2026-008",8L,"Meera Joshi","Heat Shield Pro","NO","2026-03-24","AUTO-APPROVED","HEAT",350.0,"MZ-DEL-04","2026-03-24 20:42:16"},
{"CLM-000287","POL-GW-2026-005",5L,"Vikram Singh","Multi-Peril Cover","NO","2026-03-22","AUTO-APPROVED","HEAT",350.0,"MZ-HYD-03","2026-03-24 20:41:11"},
{"CLM-000288","POL-GW-2026-007",7L,"Karan Kapoor","Heat Shield Basic","NO","2026-03-20","AUTO-APPROVED","RAIN",280.0,"MZ-PUN-02","2026-03-24 20:40:05"},
{"CLM-000201","POL-GW-2026-003",3L,"Rahul Verma","Heat Shield Basic","NO","2026-03-20","Closed","HEAT",350.0,"MZ-BLR-07","2026-03-20 10:15:00"},
};
for (Object[] d : data) {
Claim c = new Claim();
c.claimNumber = (String) d[0]; c.policyRef = (String) d[1]; c.riderId = (Long) d[2]; c.riderName = (String) d[3];
c.product = (String) d[4]; c.fraudRisk = (String) d[5]; c.dateOfLoss = (String) d[6]; c.status = (String) d[7];
c.triggerType = (String) d[8]; c.amount = (Double) d[9]; c.zone = (String) d[10]; c.approvedAt = (String) d[11];
claimRepo.save(c);
}
}
private void seedTransactions() {
Object[][] data = {
{"TXN-44021","PAYOUT","Arjun Mehta",-350.0,"2026-03-24","Parametric Claim CLM-9281 — Heat Trigger","POL-GW-2026-001"},
{"TXN-44020","PREMIUM","Deepak Reddy",125.0,"2026-03-24","Daily Premium — Multi-Peril Cover","POL-GW-2026-009"},
{"TXN-44019","PAYOUT","Priya Sharma",-280.0,"2026-03-24","Parametric Claim CLM-9284 — Rain Trigger","POL-GW-2026-002"},
{"TXN-44018","PREMIUM","Vikram Singh",112.0,"2026-03-24","Daily Premium — Multi-Peril Cover","POL-GW-2026-005"},
{"TXN-44017","PAYOUT","Sneha Patel",-350.0,"2026-03-24","Parametric Claim CLM-9282 — Heat Trigger","POL-GW-2026-004"},
{"TXN-44016","PREMIUM","Arjun Mehta",187.0,"2026-03-24","Daily Premium — Heat Shield Pro","POL-GW-2026-001"},
{"TXN-44015","PAYOUT","Anita Desai",-280.0,"2026-03-24","Parametric Claim CLM-9285 — Rain Trigger","POL-GW-2026-006"},
{"TXN-44014","PREMIUM","Sneha Patel",218.0,"2026-03-24","Daily Premium — Heat Shield Pro","POL-GW-2026-004"},
{"TXN-44013","PREMIUM","Meera Joshi",201.0,"2026-03-23","Daily Premium — Heat Shield Pro","POL-GW-2026-008"},
{"TXN-44012","PREMIUM","Lakshmi Iyer",139.0,"2026-03-23","Daily Premium — Rain Guard Plus","POL-GW-2026-010"},
{"TXN-44011","PREMIUM","Karan Kapoor",95.0,"2026-03-23","Daily Premium — Heat Shield Basic","POL-GW-2026-007"},
{"TXN-44010","PREMIUM","Rahul Verma",129.0,"2026-03-23","Daily Premium — Heat Shield Basic","POL-GW-2026-003"},
};
for (Object[] d : data) {
BillingTransaction b = new BillingTransaction();
b.txnId = (String) d[0]; b.type = (String) d[1]; b.riderName = (String) d[2]; b.amount = (Double) d[3];
b.date = (String) d[4]; b.description = (String) d[5]; b.policyRef = (String) d[6];
billingRepo.save(b);
}
}
}

File: backend/seeders/dataSeeder.js
const { Rider, Policy, Claim, BillingTransaction } = require('../models');

async function seedData() {
  const policyCount = await Policy.count();
  if (policyCount > 0) {
    console.log('[Seeder] Data already exists, skipping seed.');
    return;
  }

  console.log('[Seeder] Seeding initial data...');

  // ── Seed Riders ──
  const riderData = [
    { name: 'Arjun Mehta', phone: '+91 98765 43210', city: 'Delhi', zone: 'MZ-DEL-04', platform: 'Swiggy', age: 28 },
    { name: 'Priya Sharma', phone: '+91 87654 32109', city: 'Mumbai', zone: 'MZ-MUM-12', platform: 'Zomato', age: 31 },
    { name: 'Rahul Verma', phone: '+91 76543 21098', city: 'Bangalore', zone: 'MZ-BLR-07', platform: 'Swiggy', age: 25 },
    { name: 'Sneha Patel', phone: '+91 65432 10987', city: 'Delhi', zone: 'MZ-DEL-09', platform: 'Zepto', age: 29 },
    { name: 'Vikram Singh', phone: '+91 54321 09876', city: 'Hyderabad', zone: 'MZ-HYD-03', platform: 'Swiggy', age: 26 },
    { name: 'Anita Desai', phone: '+91 43210 98765', city: 'Chennai', zone: 'MZ-CHN-05', platform: 'Zomato', age: 30 },
    { name: 'Karan Kapoor', phone: '+91 32109 87654', city: 'Pune', zone: 'MZ-PUN-02', platform: 'Swiggy', age: 27 },
    { name: 'Meera Joshi', phone: '+91 21098 76543', city: 'Delhi', zone: 'MZ-DEL-04', platform: 'Dunzo', age: 32 },
    { name: 'Deepak Reddy', phone: '+91 10987 65432', city: 'Hyderabad', zone: 'MZ-HYD-08', platform: 'Swiggy', age: 24 },
    { name: 'Lakshmi Iyer', phone: '+91 09876 54321', city: 'Chennai', zone: 'MZ-CHN-11', platform: 'Zomato', age: 33 },
  ];

  for (const d of riderData) {
    await Rider.create({
      ...d,
      wallet_balance: 500.0,
      is_policy_active: true,
      policy_tier: 'PRO',
    });
  }

  // ── Seed Policies ──
  const policyData = [
    { policy_number: 'POL-GW-2026-001', rider_name: 'Arjun Mehta', zone: 'MZ-DEL-04 (Connaught Place)', plan: 'Heat Shield Pro', premium: '₹187/day', risk_score: 82, start_date: '2026-01-15', phone: '+91 98765 43210', email: 'arjun.mehta@gmail.com', address: '42 Janpath Road, Connaught Place, New Delhi 110001', birthdate: '1995-08-12', customer_since: '2025-11-20', account_tier: 'Gold', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-002', rider_name: 'Priya Sharma', zone: 'MZ-MUM-12 (Andheri West)', plan: 'Rain Guard Plus', premium: '₹143/day', risk_score: 71, start_date: '2026-02-03', phone: '+91 87654 32109', email: 'priya.sharma@outlook.com', address: '18/B Versova Link Road, Andheri West, Mumbai 400053', birthdate: '1992-03-25', customer_since: '2025-12-15', account_tier: 'Silver', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-003', rider_name: 'Rahul Verma', zone: 'MZ-BLR-07 (Koramangala)', plan: 'Heat Shield Basic', premium: '₹129/day', risk_score: 65, start_date: '2026-02-20', phone: '+91 76543 21098', email: 'rahul.verma@yahoo.com', address: '5th Cross, Koramangala 4th Block, Bangalore 560034', birthdate: '1998-11-07', customer_since: '2026-01-10', account_tier: 'Bronze', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-004', rider_name: 'Sneha Patel', zone: 'MZ-DEL-09 (Karol Bagh)', plan: 'Heat Shield Pro', premium: '₹218/day', risk_score: 91, start_date: '2026-01-28', phone: '+91 65432 10987', email: 'sneha.patel@gmail.com', address: '12 Pusa Road, Karol Bagh, New Delhi 110005', birthdate: '1994-06-18', customer_since: '2025-10-05', account_tier: 'Gold', delinquency_status: 'YES' },
    { policy_number: 'POL-GW-2026-005', rider_name: 'Vikram Singh', zone: 'MZ-HYD-03 (HITEC City)', plan: 'Multi-Peril Cover', premium: '₹112/day', risk_score: 58, start_date: '2026-03-01', phone: '+91 54321 09876', email: 'vikram.singh@hotmail.com', address: '301 Cyber Towers, HITEC City, Hyderabad 500081', birthdate: '1997-01-30', customer_since: '2026-02-01', account_tier: 'Bronze', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-006', rider_name: 'Anita Desai', zone: 'MZ-CHN-05 (T. Nagar)', plan: 'Rain Guard Plus', premium: '₹168/day', risk_score: 76, start_date: '2026-03-10', phone: '+91 43210 98765', email: 'anita.desai@gmail.com', address: '78 Usman Road, T. Nagar, Chennai 600017', birthdate: '1993-09-14', customer_since: '2026-01-22', account_tier: 'Silver', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-007', rider_name: 'Karan Kapoor', zone: 'MZ-PUN-02 (Hinjewadi)', plan: 'Heat Shield Basic', premium: '₹95/day', risk_score: 44, start_date: '2026-02-14', phone: '+91 32109 87654', email: 'karan.kapoor@gmail.com', address: 'Blue Ridge Township, Hinjewadi Phase 1, Pune 411057', birthdate: '1996-12-03', customer_since: '2026-01-05', account_tier: 'Bronze', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-008', rider_name: 'Meera Joshi', zone: 'MZ-DEL-04 (Connaught Place)', plan: 'Heat Shield Pro', premium: '₹201/day', risk_score: 88, start_date: '2026-01-05', phone: '+91 21098 76543', email: 'meera.joshi@outlook.com', address: '15 Barakhamba Road, Connaught Place, New Delhi 110001', birthdate: '1991-04-22', customer_since: '2025-09-18', account_tier: 'Platinum', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-009', rider_name: 'Deepak Reddy', zone: 'MZ-HYD-08 (Gachibowli)', plan: 'Multi-Peril Cover', premium: '₹125/day', risk_score: 63, start_date: '2026-03-18', phone: '+91 10987 65432', email: 'deepak.reddy@yahoo.com', address: 'ISB Road, Gachibowli, Hyderabad 500032', birthdate: '1999-07-09', customer_since: '2026-02-28', account_tier: 'Bronze', delinquency_status: 'NO' },
    { policy_number: 'POL-GW-2026-010', rider_name: 'Lakshmi Iyer', zone: 'MZ-CHN-11 (Adyar)', plan: 'Rain Guard Plus', premium: '₹139/day', risk_score: 70, start_date: '2026-02-28', phone: '+91 09876 54321', email: 'lakshmi.iyer@gmail.com', address: '22 Gandhi Nagar, Adyar, Chennai 600020', birthdate: '1990-02-16', customer_since: '2026-01-15', account_tier: 'Silver', delinquency_status: 'YES' },
  ];

  for (let i = 0; i < policyData.length; i++) {
    await Policy.create({
      ...policyData[i],
      rider_id: i + 1,
      status: 'Active',
    });
  }

  // ── Seed Claims ──
  const claimData = [
    { claim_number: 'CLM-000212', policy_ref: 'POL-GW-2026-001', rider_id: 1, rider_name: 'Arjun Mehta', product: 'Heat Shield Pro', fraud_risk: 'NO', date_of_loss: '2026-03-24', status: 'Open', trigger_type: 'HEAT', amount: 350.0, zone: 'MZ-DEL-04', approved_at: null },
    { claim_number: 'CLM-000198', policy_ref: 'POL-GW-2026-001', rider_id: 1, rider_name: 'Arjun Mehta', product: 'Heat Shield Pro', fraud_risk: 'NO', date_of_loss: '2026-03-18', status: 'Closed', trigger_type: 'HEAT', amount: 350.0, zone: 'MZ-DEL-04', approved_at: '2026-03-18 14:22:16' },
    { claim_number: 'CLM-000210', policy_ref: 'POL-GW-2026-004', rider_id: 4, rider_name: 'Sneha Patel', product: 'Heat Shield Pro', fraud_risk: 'YES', date_of_loss: '2026-03-24', status: 'Open', trigger_type: 'HEAT', amount: 350.0, zone: 'MZ-DEL-09', approved_at: null },
    { claim_number: 'CLM-000209', policy_ref: 'POL-GW-2026-002', rider_id: 2, rider_name: 'Priya Sharma', product: 'Rain Guard Plus', fraud_risk: 'NO', date_of_loss: '2026-03-24', status: 'Open', trigger_type: 'RAIN', amount: 280.0, zone: 'MZ-MUM-12', approved_at: null },
    { claim_number: 'CLM-000211', policy_ref: 'POL-GW-2026-006', rider_id: 6, rider_name: 'Anita Desai', product: 'Rain Guard Plus', fraud_risk: 'NO', date_of_loss: '2026-03-24', status: 'Open', trigger_type: 'RAIN', amount: 280.0, zone: 'MZ-CHN-05', approved_at: null },
    { claim_number: 'CLM-000207', policy_ref: 'POL-GW-2026-010', rider_id: 10, rider_name: 'Lakshmi Iyer', product: 'Rain Guard Plus', fraud_risk: 'NO', date_of_loss: '2026-03-23', status: 'Open', trigger_type: 'RAIN', amount: 280.0, zone: 'MZ-CHN-11', approved_at: null },
    { claim_number: 'CLM-000213', policy_ref: 'POL-GW-2026-008', rider_id: 8, rider_name: 'Meera Joshi', product: 'Heat Shield Pro', fraud_risk: 'NO', date_of_loss: '2026-03-24', status: 'AUTO-APPROVED', trigger_type: 'HEAT', amount: 350.0, zone: 'MZ-DEL-04', approved_at: '2026-03-24 20:42:16' },
    { claim_number: 'CLM-000287', policy_ref: 'POL-GW-2026-005', rider_id: 5, rider_name: 'Vikram Singh', product: 'Multi-Peril Cover', fraud_risk: 'NO', date_of_loss: '2026-03-22', status: 'AUTO-APPROVED', trigger_type: 'HEAT', amount: 350.0, zone: 'MZ-HYD-03', approved_at: '2026-03-24 20:41:11' },
    { claim_number: 'CLM-000288', policy_ref: 'POL-GW-2026-007', rider_id: 7, rider_name: 'Karan Kapoor', product: 'Heat Shield Basic', fraud_risk: 'NO', date_of_loss: '2026-03-20', status: 'AUTO-APPROVED', trigger_type: 'RAIN', amount: 280.0, zone: 'MZ-PUN-02', approved_at: '2026-03-24 20:40:05' },
    { claim_number: 'CLM-000201', policy_ref: 'POL-GW-2026-003', rider_id: 3, rider_name: 'Rahul Verma', product: 'Heat Shield Basic', fraud_risk: 'NO', date_of_loss: '2026-03-20', status: 'Closed', trigger_type: 'HEAT', amount: 350.0, zone: 'MZ-BLR-07', approved_at: '2026-03-20 10:15:00' },
  ];

  for (const d of claimData) {
    await Claim.create(d);
  }

  // ── Seed Billing Transactions ──
  const txnData = [
    { txn_id: 'TXN-44021', type: 'PAYOUT', rider_name: 'Arjun Mehta', amount: -350.0, date: '2026-03-24', description: 'Parametric Claim CLM-9281 — Heat Trigger', policy_ref: 'POL-GW-2026-001' },
    { txn_id: 'TXN-44020', type: 'PREMIUM', rider_name: 'Deepak Reddy', amount: 125.0, date: '2026-03-24', description: 'Daily Premium — Multi-Peril Cover', policy_ref: 'POL-GW-2026-009' },
    { txn_id: 'TXN-44019', type: 'PAYOUT', rider_name: 'Priya Sharma', amount: -280.0, date: '2026-03-24', description: 'Parametric Claim CLM-9284 — Rain Trigger', policy_ref: 'POL-GW-2026-002' },
    { txn_id: 'TXN-44018', type: 'PREMIUM', rider_name: 'Vikram Singh', amount: 112.0, date: '2026-03-24', description: 'Daily Premium — Multi-Peril Cover', policy_ref: 'POL-GW-2026-005' },
    { txn_id: 'TXN-44017', type: 'PAYOUT', rider_name: 'Sneha Patel', amount: -350.0, date: '2026-03-24', description: 'Parametric Claim CLM-9282 — Heat Trigger', policy_ref: 'POL-GW-2026-004' },
    { txn_id: 'TXN-44016', type: 'PREMIUM', rider_name: 'Arjun Mehta', amount: 187.0, date: '2026-03-24', description: 'Daily Premium — Heat Shield Pro', policy_ref: 'POL-GW-2026-001' },
    { txn_id: 'TXN-44015', type: 'PAYOUT', rider_name: 'Anita Desai', amount: -280.0, date: '2026-03-24', description: 'Parametric Claim CLM-9285 — Rain Trigger', policy_ref: 'POL-GW-2026-006' },
    { txn_id: 'TXN-44014', type: 'PREMIUM', rider_name: 'Sneha Patel', amount: 218.0, date: '2026-03-24', description: 'Daily Premium — Heat Shield Pro', policy_ref: 'POL-GW-2026-004' },
    { txn_id: 'TXN-44013', type: 'PREMIUM', rider_name: 'Meera Joshi', amount: 201.0, date: '2026-03-23', description: 'Daily Premium — Heat Shield Pro', policy_ref: 'POL-GW-2026-008' },
    { txn_id: 'TXN-44012', type: 'PREMIUM', rider_name: 'Lakshmi Iyer', amount: 139.0, date: '2026-03-23', description: 'Daily Premium — Rain Guard Plus', policy_ref: 'POL-GW-2026-010' },
    { txn_id: 'TXN-44011', type: 'PREMIUM', rider_name: 'Karan Kapoor', amount: 95.0, date: '2026-03-23', description: 'Daily Premium — Heat Shield Basic', policy_ref: 'POL-GW-2026-007' },
    { txn_id: 'TXN-44010', type: 'PREMIUM', rider_name: 'Rahul Verma', amount: 129.0, date: '2026-03-23', description: 'Daily Premium — Heat Shield Basic', policy_ref: 'POL-GW-2026-003' },
  ];

  for (const d of txnData) {
    await BillingTransaction.create(d);
  }

  console.log('[Seeder] Data seeding completed successfully.');
}

module.exports = { seedData };
